### PR TITLE
Fix: android pdf scrolling issue

### PR DESCRIPTION
**Pull request by HezekielT**
### Details

Users couldn’t scroll up and down smoothly in a pdf attachment due to a conflict between vertical(react-native-pdf’s) scroll event and horizontal(pager’s) scroll event. This PR aims to address this conflict by disabling pager’s scroll event when the scroll is in a vertical direction and enabling it when the scroll is in a horizontal direction.
### Fixed Issues

$ https://github.com/Expensify/App/issues/23417
PROPOSAL: https://github.com/Expensify/App/issues/23417#issuecomment-1762822130

### Tests

1. Open any report
2. Click on `+` of the composer.
3. Select `Add attachment` > `Choose document` > upload a multi-page pdf file.
4. Open the pdf file and scroll up and down.
5. Verify the scroll is smooth.
6. Close the attachment and Click on `+` of the composer again.
7. Upload an image.
8. Open the image and swipe to the left.
9. Verify the previously sent pdf is opened.
10. Scroll through the pdf up and down.
11. Verify the scroll is smooth.

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Make sure the uploaded pdf and image are loaded first by opening them.
2. Turn off the wifi.
3. Scroll through the pdf up and down.
4. Verify the scroll is smooth.
5. Swipe to the left/right.
6. Verify the image is opened.

### QA Steps

Same as `Tests` Section.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/39636266/27f3d4a4-94e9-44c7-94ff-0585b38e2c0a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/39636266/a2818bbf-8c96-4465-8c43-8439d67a5126

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/39636266/584ddb99-baf0-4993-9e0e-bb81aadc70d9

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/39636266/2e990e38-2b75-4093-ad82-20a569cde41c

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/39636266/0d9aa2e1-a112-4b55-a044-9f6b1106a3b3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/39636266/463d676a-fe50-4479-b42f-8097deb418e9

</details>

### CODE REVIEW

**Review by mollfpr — COMMENTED** [review]
Sorry for the ping @jjcoffee I'll review the PR 🙏

**Review by mollfpr — APPROVED** [review]
LGTM and tests well 👍

@HezekielT Could you add details on the PR description?

All yours @Gonals

File: src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js [hunk 1]
```diff
@@ -1,9 +1,34 @@
 import React, {memo, useCallback, useContext, useEffect} from 'react';
+import PropTypes from 'prop-types';
 import {attachmentViewPdfPropTypes, attachmentViewPdfDefaultProps} from './propTypes';
 import PDFView from '../../../PDFView';
 import AttachmentCarouselPagerContext from '../../AttachmentCarousel/Pager/AttachmentCarouselPagerContext';
 
-function AttachmentViewPdf({file, encryptedSourceUrl, isFocused, isUsedInCarousel, onPress, onScaleChanged: onScaleChangedProp, onToggleKeyboard, onLoadComplete, errorLabelStyles, style}) {
+const propTypes = {
+    /** Updates the scale value of the pdf */
+    updateScaleRef: PropTypes.func,
```
Comment (mollfpr) [inline src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js:9]: I don't feel this is the correct name. We usually use `ref` prefix for passing the ref.

Comment (mollfpr) [reply src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js:9]: Yeah, that sounds better.

**Review by mollfpr — APPROVED** [review]
Tested on the latest commit and still good.

Friendly bump @Gonals for final review.

**Comment by mollfpr** [comment]
@HezekielT There's a lint error, could you resolve it? Thanks!

**Comment by mollfpr** [comment]
Here's what I got in my local.

```diff
diff --git a/src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js b/src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js
index e0f3e9ab84..c56bda6972 100644
--- a/src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js
+++ b/src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js
@@ -1,5 +1,5 @@
-import React, {memo, useCallback, useContext, useEffect} from 'react';
 import PropTypes from 'prop-types';
+import React, {memo, useCallback, useContext, useEffect} from 'react';
 import AttachmentCarouselPagerContext from '@components/Attachments/AttachmentCarousel/Pager/AttachmentCarouselPagerContext';
 import PDFView from '@components/PDFView';
 import {attachmentViewPdfDefaultProps, attachmentViewPdfPropTypes} from './propTypes';
diff --git a/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.android.js b/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.android.js
index 3a5f39e188..7c3949f3f9 100644
--- a/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.android.js
+++ b/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.android.js
@@ -1,11 +1,11 @@
 import React, {memo, useCallback, useContext} from 'react';
-import {View, StyleSheet} from 'react-native';
+import {StyleSheet, View} from 'react-native';
 import {Gesture, GestureDetector} from 'react-native-gesture-handler';
 import Animated, {useSharedValue} from 'react-native-reanimated';
 import AttachmentCarouselPagerContext from '@components/Attachments/AttachmentCarousel/Pager/AttachmentCarouselPagerContext';
 import styles from '@styles/styles';
 import BaseAttachmentViewPdf from './BaseAttachmentViewPdf';
-import {attachmentViewPdfPropTypes, attachmentViewPdfDefaultProps} from './propTypes';
+import {attachmentViewPdfDefaultProps, attachmentViewPdfPropTypes} from './propTypes';

 function AttachmentViewPdf(props) {
     const attachmentCarouselPagerContext = useContext(AttachmentCarouselPagerContext);
diff --git a/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.ios.js b/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.ios.js
index 94440e1d17..103ff29276 100644
--- a/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.ios.js
+++ b/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.ios.js
@@ -1,6 +1,6 @@
 import React, {memo} from 'react';
 import BaseAttachmentViewPdf from './BaseAttachmentViewPdf';
-import {attachmentViewPdfPropTypes, attachmentViewPdfDefaultProps} from './propTypes';
+import {attachmentViewPdfDefaultProps, attachmentViewPdfPropTypes} from './propTypes';

 function AttachmentViewPdf(props) {
     return (
```

File: src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js [hunk 2]
```diff
@@ -16,6 +41,7 @@ function AttachmentViewPdf({file, encryptedSourceUrl, isFocused, isUsedInCarouse
 
     const onScaleChanged = useCallback(
         (scale) => {
+            updateScale(scale);
```
Comment (mollfpr) [inline src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js:44]: Looking again at these props, I think we can just use `onScaleChanged` to pass the scale down, and then override it in the

```diff
 function AttachmentViewPdf(props) {
+    const {onScaleChanged, ...restProps} = props;
     const attachmentCarouselPagerContext = useContext(AttachmentCarouselPagerContext);
     const scaleRef = useSharedValue(1);
     const offsetX = useSharedValue(0);
@@ -49,8 +50,11 @@ function AttachmentViewPdf(props) {
                 >
                     <BaseAttachmentViewPdf
                         // eslint-disable-next-line react/jsx-props-no-spreading
-                        {...props}
-                        updateScale={updateScale}
+                        {...restProps}
+                        onScaleChanged={(scale) => {
+                            updateScale(scale)
+                            onScaleChanged()
+                        }}
                     />
                 </Animated.View>
             </GestureDetector>
```

```diff
-            onScaleChangedProp();
+            onScaleChangedProp(scale);
```

**Review by mollfpr — APPROVED** [review]

**Review by Gonals — APPROVED** [review]

**Comment by situchan** [comment]
This PR caused a few deploy blockers: #30977 #30979

**Comment by mollfpr** [comment]
@HezekielT Could you check the issue?

**Comment by akinwale** [comment]
This PR caused a [regression](https://github.com/Expensify/App/issues/49761) on Android native where the PDF being zoomed in isn't taken into account when attempting to scroll left / right to change the currently viewed attachment.

**Pull request merged**